inc/classes/class-logger.php

```diff
@@ -0,0 +1,479 @@
+	 * @return bool|null       Whether the record has been processed.
+	 */
+	public static function critical( $message, array $context = array() ) {
+		return static::debug_enabled() ? static::get_logger()->critical( $message, $context ) : null;
+	}
+
+	/**
+	 * Adds a log record at the ALERT level.
+	 *
+	 * @since  3.1.4
+	 * @access public
+	 * @author Grégory Viguier
+	 *
+	 * @param  string $message The log message.
+	 * @param  array  $context The log context.
+	 * @return bool|null       Whether the record has been processed.
+	 */
+	public static function alert( $message, array $context = array() ) {
+		return static::debug_enabled() ? static::get_logger()->alert( $message, $context ) : null;
+	}
+
+	/**
+	 * Adds a log record at the EMERGENCY level.
+	 *
+	 * @since  3.1.4
+	 * @access public
+	 * @author Grégory Viguier
+	 *
+	 * @param  string $message The log message.
+	 * @param  array  $context The log context.
+	 * @return bool|null       Whether the record has been processed.
+	 */
+	public static function emergency( $message, array $context = array() ) {
+		return static::debug_enabled() ? static::get_logger()->emergency( $message, $context ) : null;
+	}
+
+	/**
+	 * Get the logger instance.
+	 *
+	 * @since  3.1.4
+	 * @access public
+	 * @author Grégory Viguier
+	 *
+	 * @return Logger A Logger instance.
+	 */
+	public static function get_logger() {
+		$logger_name = static::LOGGER_NAME;
+		$log_level   = Monologger::DEBUG;
+
+		if ( Registry::hasLogger( $logger_name ) ) {
+			return Registry::$logger_name();
+		}
+
+		/**
+		 * File handler.
+		 * HTML formatter is used.
+		 *
+		 * Note, if Inline formatter is used:
+		 *     - Keep default formats, allow line breaks in messages, hide empty contexts and extras.
+		 *     - `$formatter = new LineFormatter( null, null, true, true );`.
+		 */
+		$handler   = new StreamHandler( static::get_log_file_path(), $log_level );
+		$formatter = new HtmlFormatter();
+
+		$handler->setFormatter( $formatter );
+
+		/**
+		 * Thanks to the processors, add data to each log:
+		 * - `debug_backtrace()` (exclude this class).
+		 */
+		$trace_processor = new Processor\IntrospectionProcessor( $log_level, [ get_called_class() ] );
+
+		// Create the logger.
+		$logger = new Monologger( $logger_name, [ $handler ], [ $trace_processor ] );
+
+		// Store the logger.
+		Registry::addLogger( $logger );
+
+		return $logger;
+	}
+
+
+	/** ----------------------------------------------------------------------------------------- */
+	/** LOG FILE ================================================================================ */
+	/** ----------------------------------------------------------------------------------------- */
+
+	/**
+	 * Get the path to the log file.
+	 *
+	 * @since  3.1.4
+	 * @access public
+	 * @author Grégory Viguier
+	 *
+	 * @return string
+	 */
+	public static function get_log_file_path() {
+		return WP_CONTENT_DIR . '/wp-rocket-config/' . static::LOG_FILE_NAME;
```

Comment:
I think we can add a filter here if someone wants to modify the path

Reply:
I thought about it but we can't: this method is used in the method `get_logger()`, which is used in the file `process.php`, which is not in WordPress environment, so `apply_filters()` won't exist.